py3status/modules/xrandr.py

```diff
@@ -141,17 +156,29 @@ def _set_available_combinations(self):
         available_combinations = set()
         combinations_map = {}
 
+        if self.output_combinations is None:
+            whitelist = None
+        else:
+            whitelist = self.output_combinations.split('|')
+
         self.max_width = 0
-        for output in range(len(self.layout['connected']) + 1):
+        for output in range(1, len(self.layout['connected']) + 1):
             for comb in combinations(self.layout['connected'], output):
-                if comb:
-                    for mode in ['clone', 'extend']:
-                        string = self._get_string_and_set_width(comb, mode)
-                        if len(comb) == 1:
-                            combinations_map[string] = (comb, None)
-                        else:
-                            combinations_map[string] = (comb, mode)
-                        available_combinations.add(string)
+                for mode in ['clone', 'extend']:
+                    string = self._get_string_and_set_width(comb, mode)
+                    if whitelist and string not in whitelist:
+                        continue
+                    if len(comb) == 1:
+                        combinations_map[string] = (comb, None)
+                    else:
+                        combinations_map[string] = (comb, mode)
+                    available_combinations.add(string)
+
+        # Preserve the order in which user defined the output combinations
+        if whitelist is not None:
+            available_combinations = [comb for comb in whitelist
+                                      if comb in available_combinations][::-1]
```

Comment:
Why? I think list slicing is a very pythonic thingy, it becomes quite clear as soon as you get used to it - I actually miss this simplicity in many other languages :)

We inverse because the existing code that scrolls through available combinations on mouse click is doing that in inverse order, and I didn't want to change that part of a code not to affect existing users that are already accustomed to this.

I did put a general comment that this part of code does the correct order, but not explained in details as it might get complicated.